decidim-admin/app/views/decidim/admin/components/_component.html.erb

```diff
@@ -34,7 +34,11 @@
         <% if component.published? %>
           <%= icon_link_to "x", url_for(action: :unpublish, id: component, controller: "components"), t("actions.unpublish", scope: "decidim.admin"), class: "action-icon--unpublish", method: :put %>
         <% else %>
-          <%= icon_link_to "check", url_for(action: :publish, id: component, controller: "components"), t("actions.publish", scope: "decidim.admin"), class: "action-icon--publish", method: :put %>
+          <% if component.manifest_name == "surveys" && component.settings.clean_after_publish %>
```

Comment:
When trying to merge, this change will result in a conflict. 

As stated [before](https://github.com/decidim/decidim/pull/11423#discussion_r1336413728), i am not a big fan for binding modules together just for additional checks. 
I would extract the component actions line 15 to end in a separate partial ( named actions ), then use something like: 

```
if lookup_context.find_all("decidim/#{component.manifest_name}/admin/component/actions").any?
  render partial: "decidim/#{component.manifest_name}/admin/component/actions"
else 
 render partial: "actions"
end 
```

Then create a new `_action.html.erb` partial in the following path  `decidim-surveys/app/views/decidim/surveys/admin/component/_actions.html.erb` that would carry the specific survey related checks.

Reply:
hello @alecslupu , I have a doubt: 

The partial "actions" in else condition, which is it?

Reply:
I have made some changes related to this, can you take a look at them?